src/watcher.js

```diff
@@ -266,6 +271,29 @@ class Watcher extends SafeEventEmitter {
         this.on('load.chat', updateChannel);
         this.on('load.vod', updateChannel);
     }
+
+    directoryObserver() {
+        const observe = (watcher, element) => {
+            if (!element) return;
+            if (watcher) watcher.disconnect();
+            watcher.observe(element, {childList: true, subtree: true});
+        };
+
+        directoryWatcher = new window.MutationObserver(mutations =>
+            mutations.forEach(mutation => {
+                for (const el of mutation.addedNodes) {
+                    const $el = $(el);
+
+                    const vodCasts = $el.find('.tw-pill');
+
+                    if (vodCasts.length > 0) {
+                        this.emit('directory.vodcast');
```

Comment:
this should pass an element .. not rely on `this`

Reply:
Changed vodCasts to $vodCasts and added that to this.emit: this.emit('directory.vodcast', $vodCasts);